lib/Doctrine/Common/Persistence/Mapping/Driver/AnnotationDriver.php

```diff
@@ -253,4 +249,43 @@ public function getAllClassNames()
 
         return $classes;
     }
+
+    /**
+     * Returns the canonicalized absolute pathname. This method is a Phar-safe version
+     * of the builtin `realpath()` function.
+     *
+     * @param string $path
+     *
+     * @return boolean|string
+     */
+    protected function realpath($path)
+    {
+        if ('phar://' !== substr($path, 0, 7)) {
```

Comment:
the phar stream wrapper is not the only one not supporting realpath. Otherwise stream wrappers might suffer the same.

What you should do is call realpath, and do the fallback logic when it returns ``false`` (and make the fallback logic support other cases than the phar stream wrapper)